### PR TITLE
nit: allow arbitrary folder recursion for e2e tests.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Something I noticed in https://github.com/cosmos/ibc-go/pull/3710. In addition to that, doesn't seem like `03-connection` tests are currently checked during the `build` step (though their tests are executed afterwards).

closes: #XXXX

### Commit Message / Changelog Entry

```bash
nit: allow arbitrary folder recursion for e2e tests.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -50,10 +50,11 @@ jobs:
         - name: Build e2e
           run: |
             cd e2e
-            test_dirs="$(ls -A tests)"
-            for td in $test_dirs
+            find ./tests -type d | while IFS= read -r dir
```
Comment (DimitrisJim) [inline .github/workflows/e2e.yaml:53]: finds all folders in `tests/` and if they contain a `.go` file, compiles the test binary

Comment (damiannolan) [reply .github/workflows/e2e.yaml:53]: So just so I understand this pipe - IFS = input field separator, and its separating on blank/space(?) reading from stdin into var `dir` which we then use inside the loop

Comment (DimitrisJim) [reply .github/workflows/e2e.yaml:53]: typically it is used with some separator value if required, setting it to nothing `IFS=` means it basically won't do any further splitting of the file paths (allowing e.g a path with spaces if one arises). A ref: https://unix.stackexchange.com/a/676870

**Review by damiannolan — APPROVED** [review]
lgtm 🚀

**Review by charleenfei — APPROVED** [review]
nice fix!

**Review by chatton — APPROVED** [review]
LGTM great improvement ❤️

**Pull request merged**